cli/src/infragpt/llm/providers/openai_provider.py

```
@@ -0,0 +1,241 @@
+                    
+                    # Only emit tool calls once when we get the finish reason
+                    if finish_reason == "tool_calls" and accumulated_tool_calls:
+                        yield StreamChunk(
+                            content=None,
+                            tool_calls=accumulated_tool_calls,
+                            finish_reason=finish_reason
+                        )
+                        accumulated_tool_calls = []  # Reset after emitting
+                    
+        except Exception as e:
+            raise self._map_error(e)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)